docs/configuration-format.md

```diff
@@ -0,0 +1,152 @@
+# OpenTelemetry Collector Configuration Format
+
+The effective configuration that defines runtime behavior of the collector is calculated from 2 sources: the Default configuration which is hard-coded and the User-defined configuration which supplied by the user via various configuration sources. Default configuration defines reasonable defaults that most users will likely want to use. User-defined configuration can override the defaults and specify additional configuration. The effective configuration is calculated by [merging Default and User-defined configurations](#merging-rules).
```

Comment:
We need to reword this. This was the original intent but we didn't implement it that way. There is no "default" full config that is merged with user config. We only have per-component defaults.